:memo: Getting started: OS X and DNX

natemcmaster opened the pull request:
#4 

Should be available here http://ef.readthedocs.org/en/staging/getting-started/osx.html

The sample likely have some build and execution errors until DNX renames are finished. 

Please help me check for clarity, grammar, etc.

dnfclas commented:
Hi __@natemcmaster__, I'm your friendly neighborhood .NET Foundation Pull Request Bot (You can call me DNFBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by .NET Foundation and real humans are currently evaluating your PR.</p>

TTYL, DNFBOT;

dnfclas commented:
Hi __@natemcmaster__, I'm your friendly neighborhood .NET Foundation Pull Request Bot (You can call me DNFBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by .NET Foundation and real humans are currently evaluating your PR.</p>

TTYL, DNFBOT;

ErikEJ commented on docs/getting-started/osx.rst:
```diff
@@ -1,8 +1,187 @@
+.. note::
+    **Known Issues**
+
+     - Bugs in Mono 4.0.2 may cause Entity Framework to crash when using async methods. This is resolve with Mono >4.2.0, which has not yet been publicly released. `See this issue on GitHub <https://github.com/aspnet/EntityFramework/issues/2708>`_
```
resolve => resolved

ErikEJ commented on docs/getting-started/osx.rst:
```diff
@@ -1,8 +1,187 @@
+blog post from the command line.
+
+ 1. Create a new class to represent the SQLite database. We will call this ``BloggingContext``. Note that to configure this for SQLite, we must call ``UseSqlite()`` with a connection string
+    pointing to the \*.db file.
```
Forward slash or backslash?

natemcmaster replied:
This is just escaping the * for rst files.

natemcmaster commented:
Fixed some typos, updated formatting, and simplified the sample program.

Also, I made the installation commands more clear and added a final step to make it more clear how to execute the app.

rowanmiller commented on docs/getting-started/osx.rst:
```diff
@@ -1,8 +1,230 @@
+.. note:: `View this article's samples on GitHub <https://github.com/aspnet/EntityFramework.Docs/tree/master/docs/getting-started/x-plat/sample>`_.
+
+You can find nightly builds of the EF7 code base hosted on https://www.myget.org/F/aspnetvnext/api/v2/ but the code base is rapidly changing and we do not maintain up-to-date documentation for getting started.
+
```
I think line 22 should be part of the note from line 9

rowanmiller commented on docs/getting-started/osx.rst:
```diff
@@ -1,8 +1,230 @@
+ - Mono 4.0.2
+ - OS X Mavericks
+
+.. note::
```
use a `.. caution::`

rowanmiller commented on docs/getting-started/osx.rst:
```diff
@@ -1,8 +1,230 @@
+
+
+Begin using EF
+--------------
```
Rename title to "Create your model" (to align with other walkthroughs)

rowanmiller commented on docs/getting-started/osx.rst:
```diff
@@ -1,8 +1,230 @@
+We will create a simple console application that allows us to write a 
+blog post from the command line.
+
+ - Create a new file called ``BloggingModel.cs``
```
Call this Model.cs and just list the entire code in one block (to align with other walkthroughs).
http://ef.readthedocs.org/en/latest/getting-started/full-dotnet.html#create-your-model

rowanmiller commented on docs/getting-started/osx.rst:
```diff
@@ -1,8 +1,230 @@
+        Total projects built: 1
+
+Create the database with migrations
+-----------------------------------
```
Rename title to "Create your database"

rowanmiller commented on docs/getting-started/osx.rst:
```diff
@@ -1,8 +1,230 @@
+
+    .. code-block:: console
+
+        $ dnx . ef migration add SampleInitial
```
Call it "MyFirstMigration" to align with other walkthroughs

rowanmiller commented on docs/getting-started/osx.rst:
```diff
@@ -1,8 +1,230 @@
+    This should create a new file, ``blog.db`` which contains two empty tables.
+
+Use the context to access data
+------------------------------
```
Rename to "Use your model"

rowanmiller commented on docs/getting-started/x-plat/sample/src/ConsoleApp/Program.cs:
```diff
@@ -0,0 +1,38 @@
+            }
+        }
+    }
+}
```
This code block should be consistent with the other walkthroughs (unless there is a need to change it for OSX).

```
            using (var db = new BloggingContext())
            {
                db.Blogs.Add(new Blog { Url = "http://blogs.msdn.com/adonet" });
                var count = db.SaveChanges();
                Console.WriteLine("{0} records saved to database", count);

                Console.WriteLine();
                Console.WriteLine("All blogs in database:");
                foreach (var blog in db.Blogs)
                {
                    Console.WriteLine(" - {0}", blog.Url);
                }
            }
```

rowanmiller commented:
:shipit:

rowanmiller commented:
BTW the commit emoji should be `:memo:` not :menu: :smile:

natemcmaster commented:
Responded to comments and merged
